### PR TITLE
feat(dashboard): add SSO feature and integrate with LDAP

### PR DESCRIPTION
**Pull request by lafirest**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at 2cddce5</samp>

This pull request adds a new application `emqx_dashboard_sso` to the enterprise edition of EMQX, which provides the functionality of single sign-on for the EMQX dashboard. It implements different single sign-on backends, such as LDAP, and integrates them with the existing `emqx_dashboard` and `emqx_dashboard_admin` modules. It also adds the necessary configuration, documentation, license, and internationalization files for the new application.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by zhongwencool — APPROVED** [review]

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_api.erl [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+%% API
+login(post, #{bindings := #{backend := Backend}, body := Sign}) ->
+    case emqx_dashboard_sso_manager:lookup_state(Backend) of
+        undefined ->
+            {404, ?BACKEND_NOT_FOUND, <<"Backend not found">>};
+        State ->
+            Provider = emqx_dashboard_sso:provider(Backend),
+            case emqx_dashboard_sso:login(Provider, Sign, State) of
+                {ok, Token} ->
+                    ?SLOG(info, #{msg => "Dashboard SSO login successfully", request => Sign}),
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_api.erl:124]: dashboard_sso_login_succeeded

File: apps/emqx_dashboard/src/emqx_dashboard_api.erl [hunk 2]
```diff
@@ -206,7 +207,10 @@ field(old_pwd) ->
 field(new_pwd) ->
     {new_pwd, mk(binary(), #{desc => ?DESC(new_pwd)})};
 field(role) ->
-    {role, mk(binary(), #{desc => ?DESC(role), example => ?ROLE_DEFAULT})}.
+    {role,
+        mk(binary(), #{desc => ?DESC(role), default => ?ROLE_DEFAULT, example => ?ROLE_DEFAULT})};
+field(backend) ->
+    {backend, mk(binary(), #{desc => ?DESC(backend), example => <<"local">>})}.
```
Comment (zmstone) [inline apps/emqx_dashboard/src/emqx_dashboard_api.erl:213]: enum?

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_ldap.erl [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        {error, _} = Error ->
+            Error
+    end.
+
+parse_config(Config) ->
+    State = lists:foldl(
+        fun(Key, Acc) ->
+            case maps:find(Key, Config) of
+                {ok, Value} when is_binary(Value) ->
+                    Acc#{Key := erlang:binary_to_list(Value)};
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_ldap.erl:123]: why is 'timeout' a s string?
maybe add a comment here.

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+-record(dashboard_sso, {
+    backend :: atom(),
+    state :: map()
+}).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+start_link() ->
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:55]: maybe no need to be a gen_server.

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+%%------------------------------------------------------------------------------
+%% gen_server callbacks
+%%------------------------------------------------------------------------------
+init([]) ->
+    process_flag(trap_exit, true),
+    emqx_conf:add_handler(?MOD_KEY_PATH, ?MODULE),
+    emqx_utils_ets:new(
+        dashboard_sso,
+        [
+            set,
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:115]: ordered_set.

**Pull request merged**